### PR TITLE
Select algo according to json config

### PR DESCRIPTION
**Pull request by Binyang2014**
The way to run nccl-test over mscclpp:
mpirun -np 8 --bind-to numa --allow-run-as-root -x LD_PRELOAD=$(pwd)/build/apps/nccl/libmscclpp_nccl.so -x NCCL_DEBUG=WARN  -x MSCCLPP_EXECUTION_PLAN_DIR=/execution-files /root/nccl-tests/build/all_reduce_perf   -b 1K -e 1G -f 2 -d half -G 20 -w 10 -n 20

### CODE REVIEW

File: src/include/execution_plan.hpp [hunk 1]
```diff
@@ -63,6 +63,7 @@ struct ChannelInfo {
 struct ExecutionPlan::Impl {
  public:
   Impl(const std::string name, const std::string planPath);
```
Comment (chhwang) [inline src/include/execution_plan.hpp:65]: Looks like we should remove this line

File: apps/nccl/src/nccl.cu [hunk 2]
```diff
@@ -183,6 +192,13 @@ static std::vector<mscclpp::SmChannel> setupSmChannels(ncclComm_t comm,
   return channels;
 }
 
+static std::pair<std::string, executionPlanInstance> loadExecutionPlan(const char* filename) {
```
Comment (chhwang) [inline apps/nccl/src/nccl.cu:195]: no need to make it `const char *`, better to keep it `const std::string&`

File: apps/nccl/src/nccl.cu [hunk 3]
```diff
@@ -383,52 +399,16 @@ NCCL_API ncclResult_t ncclCommInitRank(ncclComm_t* comm, int nranks, ncclUniqueI
...
-  else
-    commPtr->allGatherLargeMessageSizeBoundary = 1 << 20;
-
-  if (commPtr->allReduceSmallMessageSizeBoundary > commPtr->allReduceLargeMessageSizeBoundary)
-    return ncclInvalidArgument;
-  if (commPtr->allGatherSmallMessageSizeBoundary > commPtr->allGatherLargeMessageSizeBoundary)
-    return ncclInvalidArgument;
+  if (getenv("MSCCLPP_EXECUTION_PLAN_DIR")) {
+    std::string collectiveDir = getenv("MSCCLPP_EXECUTION_PLAN_DIR");
+    for (const auto& entry : std::filesystem::directory_iterator(collectiveDir)) {
```
Comment (chhwang) [inline apps/nccl/src/nccl.cu:404]: Need an exception handler in case `collectiveDir` is not a valid directory

**Review by chhwang — APPROVED** [review]
LGTM

File: apps/nccl/src/nccl.cu (hunk 3 above)
Comment (Binyang2014) [reply apps/nccl/src/nccl.cu:407]: Thanks, check with `std::filesystem::is_directory(collectiveDir)` first and return ncclInvalidArgument if dir is invalid

File: apps/nccl/src/nccl.cu [hunk 4]
```diff
@@ -383,52 +399,19 @@ NCCL_API ncclResult_t ncclCommInitRank(ncclComm_t* comm, int nranks, ncclUniqueI
...
-  if (commPtr->allGatherSmallMessageSizeBoundary > commPtr->allGatherLargeMessageSizeBoundary)
-    return ncclInvalidArgument;
+  if (getenv("MSCCLPP_EXECUTION_PLAN_DIR")) {
+    std::string collectiveDir = getenv("MSCCLPP_EXECUTION_PLAN_DIR");
+    if (!std::filesystem::is_directory(collectiveDir)) {
+      return ncclInvalidArgument;
+    }
+    for (const auto& entry : std::filesystem::directory_iterator(collectiveDir)) {
+      if (entry.is_regular_file()) {
+        std::string filename = entry.path().filename().string();
```
Comment (caiomcbr) [inline apps/nccl/src/nccl.cu:409]: Are you using this variable?

Comment (Binyang2014) [reply apps/nccl/src/nccl.cu:409]: removed it. Thanks

**Pull request merged**